pkg/validation/test.go

```diff
@@ -398,67 +398,17 @@ func validateTestStepDependencies(config *api.ReleaseBuildConfiguration) []error
 	return errs
 }
 
-func validateClusterProfile(fieldRoot string, p api.ClusterProfile) []error {
-	switch p {
-	case
-		api.ClusterProfileAWS,
-		api.ClusterProfileAWSArm64,
-		api.ClusterProfileAWSAtomic,
-		api.ClusterProfileAWSCentos,
-		api.ClusterProfileAWSCentos40,
-		api.ClusterProfileAWSQE,
-		api.ClusterProfileAWSC2SQE,
-		api.ClusterProfileAWSChinaQE,
-		api.ClusterProfileAWSGovCloudQE,
-		api.ClusterProfileAWSSC2SQE,
-		api.ClusterProfileAWSGluster,
-		api.ClusterProfileAWSOSDMSP,
-		api.ClusterProfileAlibabaCloud,
-		api.ClusterProfileAzure2,
-		api.ClusterProfileAzure4,
-		api.ClusterProfileAzureArc,
-		api.ClusterProfileAzureStack,
-		api.ClusterProfileAzureMag,
-		api.ClusterProfileAzureQE,
-		api.ClusterProfileAzureMagQE,
-		api.ClusterProfileEquinixOcpMetal,
-		api.ClusterProfileGCPQE,
-		api.ClusterProfileGCP,
-		api.ClusterProfileGCP2,
-		api.ClusterProfileGCP40,
-		api.ClusterProfileGCPHA,
-		api.ClusterProfileGCPCRIO,
-		api.ClusterProfileGCPLogging,
-		api.ClusterProfileGCPLoggingJournald,
-		api.ClusterProfileGCPLoggingJSONFile,
-		api.ClusterProfileGCPLoggingCRIO,
-		api.ClusterProfileIBMCloud,
-		api.ClusterProfileLibvirtPpc64le,
-		api.ClusterProfileLibvirtS390x,
-		api.ClusterProfileNutanix,
-		api.ClusterProfileOpenStack,
-		api.ClusterProfileOpenStackKuryr,
-		api.ClusterProfileOpenStackNFV,
-		api.ClusterProfileOpenStackMechaCentral,
-		api.ClusterProfileOpenStackMechaAz0,
-		api.ClusterProfileOpenStackOsuosl,
-		api.ClusterProfileOpenStackVexxhost,
-		api.ClusterProfileOpenStackPpc64le,
-		api.ClusterProfileOvirt,
-		api.ClusterProfilePacket,
-		api.ClusterProfilePacketAssisted,
-		api.ClusterProfilePacketSNO,
-		api.ClusterProfileVSphere,
-		api.ClusterProfileVSphereDiscon,
-		api.ClusterProfileVSphereClusterbot,
-		api.ClusterProfileVSpherePlatformNone,
-		api.ClusterProfileVSphereMultizone,
-		api.ClusterProfileKubevirt,
-		api.ClusterProfileAWSCPaaS,
-		api.ClusterProfileAWS2,
-		api.ClusterProfileOSDEphemeral,
-		api.ClusterProfileHyperShift:
-		return nil
+func (v *Validator) validateClusterProfile(fieldRoot string, p api.ClusterProfile) []error {
+	if v.validClusterProfiles != nil {
+		if _, ok := v.validClusterProfiles[p]; ok {
+			return nil
+		}
+	} else {
+		for _, x := range api.ClusterProfiles() {
+			if x == p {
+				return nil
+			}
+		}
 	}
```

Comment:
@droslean: sorry for the abrupt push, the only difference to the version you approved was moving the `for` into the `else` here (a (des-des-)optimization, not a functional change).